### PR TITLE
Change how we setup travis to use composer to install all dependencies

### PR DESCRIPTION
**Pull request by jonathangreen**
The way were we installing Drupal for travis before was differnt from
how we installed it in the vagrant box. It required manually adding
dependancies for libraries. This should install all the dependancies
using composer automatically.

Came up on this pull request:
https://github.com/Islandora-CLAW/islandora/pull/38

Figured I should break it out to another pull since it got larger then 
I intended it to be.

### CODE REVIEW

**Comment by jonathangreen** [comment]
After this lands, https://github.com/Islandora-CLAW/islandora_collection should probably get the same treatment.

File: .travis.yml [hunk 1]
```diff
@@ -4,32 +4,30 @@ php:
...
+  - cd web; drush en -y islandora
 
 script:
-  - $ISLANDORA_DIR/.scripts/line_endings.sh .
-  - cd $HOME/drupal/modules/islandora
-  - php vendor/bin/phpcs --standard=Drupal --extensions=module/php,php/php src tests
-  - php vendor/bin/phpcpd --names *.module,*.inc,*.test,*.php --exclude=vendor .
-  - cd $HOME/drupal; php core/scripts/run-tests.sh --verbose --php `which php` islandora
+  - $SCRIPT_DIR/line_endings.sh $TRAVIS_BUILD_DIR
+  - phpcs --standard=Drupal $TRAVIS_BUILD_DIR
```
Comment (whikloj) [inline .travis.yml]: I'm not sure we need the old `--extensions=module/php,php/php`, but as the default is 
> ...check any file it finds with a .inc, .php, .js or .css extension, ...

Should we just specify `--extensions=php`?

Comment (jonathangreen) [inline .travis.yml]: good catch. probably not checking enough file types. should call it with the extensions listed here:
https://www.drupal.org/node/1587138

`phpcs --standard=Drupal --extensions=php,module,inc,install,test,profile,theme,css,info,txt,md /file/to/drupal/example_module`

Will update.

Comment (jonathangreen) [inline .travis.yml]: Pushed an update to test all the file types.

**Comment by jonathangreen** [comment]
@DiegoPino I feel like if we want to target a particular version of drupal we should probably do it here: 
https://github.com/Islandora-CLAW/drupal-project/blob/8.x-1.x/composer.json#L23

So it applies to both travis and the vagrant build.

File: README.md [hunk 2]
```diff
@@ -16,10 +16,22 @@ Current maintainers:
...
 
 [GPLv2](http://www.gnu.org/licenses/gpl-2.0.txt)
+
+[1]: https://travis-ci.org/Islandora-CLAW/islandora.png?branch=8.x-1.x
+[2]: http://img.shields.io/badge/CONTRIBUTING-Guidelines-blue.svg
+[3]: https://img.shields.io/badge/license-GPLv2-blue.svg?style=flat-square
+[4]: https://github.com/Islandora-CLAW/CLAW/wiki
+[5]: http://islandora.ca/sites/default/files/islandora_cla.pdf
+[6]: http://islandora.ca/sites/default/files/islandora_ccla.pdf
+[7]: http://islandora.ca/resources/contributors
```
Comment (whikloj) [inline README.md]: Is this a new feature of Github Markdown? 'Cause this is awesome

**Comment by whikloj** [comment]
This is self-testing in that we know islandora passes with the existing Travis setup and you can see how it runs with the new Travis setup. 👍  merging.

File: README.md (hunk 2 above)
Comment (jonathangreen) [inline README.md]: Seems like its always been a feature. I'd never actually seen it in use before I went looking for a way to make the readme a little more readable though.

**Comment by DiegoPino** [comment]
@jonathangreen are you making pulls for https://github.com/islandora-claw/islandora_collection and https://github.com/islandora-claw/claw-jsonld to level-up/match Travis-CI to this one? Need help on those?

**Comment by jonathangreen** [comment]
Sure. I'll work on it. I made a new issue here so I don't forget:
https://github.com/Islandora-CLAW/CLAW/issues/540

Can someone assign it to me? Thanks.

**Pull request merged**